vllm/engine/llm_engine.py

```diff
@@ -1643,9 +1663,25 @@ def _get_stats(self,
         if self.lora_config:
             max_lora_stat = str(self.lora_config.max_loras)
 
+        # Calculate total tokens in queue
+        total_tokens_in_queue = 0
+        for scheduler in self.scheduler:
+            waiting_queue = scheduler.waiting
+            for waiting_seq_group in waiting_queue:
```

Comment:
I commented this on the other PR.

We cannot have a loop over `scheduler.waiting` in every step of `llm_engine` because the waiting queue can be of arbitrary size (especially in offline batch use cases). This is a massive performance footgun.